### DISCUSSION TITLE
Navigation Error : ArgumentException: Region with the given name is already registered: ContentRegion

### DISCUSSION THREAD

**Discussion by danielnoh0514** [PrismLibrary/Prism#3070] [WPF] [answered]
![image](https://github.com/PrismLibrary/Prism/assets/118356044/b8a3b9b3-91c5-46cd-9dd0-665a855f813d)
reservationlistingview is the window and makereservationview is the usercontrol

![image](https://github.com/PrismLibrary/Prism/assets/118356044/5efb1d60-14c0-4d87-94c7-335c903db2ba)
i made a contentregion in the reservationlistingview

![image](https://github.com/PrismLibrary/Prism/assets/118356044/38ff8da0-66e9-43a5-aec5-967335fab321)
and i registered the views here

_regionManager.RequestNavigate("ContentRegion", "ReservationListingView");
  _regionManager.RequestNavigate("ContentRegion", "MakeReservationView");
i simply used requestnavigate, but it's causing the error as the screenshot below
![image](https://github.com/PrismLibrary/Prism/assets/118356044/8e6ca682-a96f-4154-847f-f97e46bbf99e)

do i need to use inavigationaware interface and remove the view or something?

**Comment by brianlagunas** [comment, marked as answer]
Hard to say without a reproduction app, this this error is telling that you have the a region with the same name of ContentRegion created twice in your app. Double check your region names and app structure. Also, you don't need to register your ReservationListingView for navigation, because it's your shell. You only register things for navigation if they will be participating in region navigation.

**Reply by danielnoh0514** [reply]
Thank you for your answer. By the way, is it better (or more common) to make other views as modules and leave the shell alone in a single project? I'm barely seeing anyone making a view in the same project where the shell is. 

Thanks

**Reply by brianlagunas** [reply]
It's more common to place your views into separate modules. The Shell application is usually just that... the Shell of the application.
